app/platform/feature_flags.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package platform
+
+import (
+	"encoding/json"
+	"os"
+	"time"
+
+	"github.com/mattermost/mattermost-server/v6/app/featureflag"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+// SetupFeatureFlags called on startup and when the cluster leader changes.
+// Starts or stops the synchronization of feature flags from upstream management.
+func (ps *PlatformService) SetupFeatureFlags() {
+	ps.featureFlagSynchronizerMutex.Lock()
+	defer ps.featureFlagSynchronizerMutex.Unlock()
+	splitKey := *ps.Config().ServiceSettings.SplitKey
+	splitConfigured := splitKey != ""
+	syncFeatureFlags := splitConfigured && ps.IsLeader()
+
+	ps.configStore.SetReadOnlyFF(!splitConfigured)
+
+	if syncFeatureFlags {
+		if err := ps.startFeatureFlagUpdateJob(); err != nil {
+			mlog.Warn("Unable to setup synchronization with feature flag management. Will fallback to cache.", mlog.Err(err))
+		}
+	} else {
+		ps.StopFeatureFlagUpdateJob()
+	}
+
+	if err := ps.configStore.Load(); err != nil {
+		mlog.Warn("Unable to load config store after feature flag setup.", mlog.Err(err))
+	}
+}
+
+func (ps *PlatformService) updateFeatureFlagValuesFromManagement() {
+	newCfg := ps.configStore.GetNoEnv().Clone()
+	oldFlags := *newCfg.FeatureFlags
+	newFlags := ps.featureFlagSynchronizer.UpdateFeatureFlagValues(oldFlags)
+	oldFlagsBytes, _ := json.Marshal(oldFlags)
+	newFlagsBytes, _ := json.Marshal(newFlags)
+	mlog.Debug("Checking feature flags from management service", mlog.String("old_flags", string(oldFlagsBytes)), mlog.String("new_flags", string(newFlagsBytes)))
+	if oldFlags != newFlags {
+		mlog.Debug("Feature flag change detected, updating config")
+		*newCfg.FeatureFlags = newFlags
+		ps.SaveConfig(newCfg, true)
+	}
+}
+
+func (ps *PlatformService) startFeatureFlagUpdateJob() error {
+	// Can be run multiple times
+	if ps.featureFlagSynchronizer != nil {
+		return nil
+	}
+
+	var log *mlog.Logger
+	if *ps.Config().ServiceSettings.DebugSplit {
+		var err error
+		// TODO: Add split logging functionality back and probably use PlatformService.Logger once added.
+		log, err = mlog.NewLogger()
+		if err != nil {
+			return err
+		}
```

Comment:
We already have a `ps.logger` field right? Any reason we can't use that?

Reply:
Good catch 👍